packages/sitecore-jss-nextjs/src/services/graphql-sitemap-service.ts

```diff
@@ -61,225 +37,116 @@ export interface GraphQLSitemapServiceConfig {
    * The API key to use for authentication.
    */
   apiKey: string;
```

Comment:
Feel like there's an opportunity to create a base config that includes 'endpoint' and 'apiKey' as well. `GraphQLServiceConfig`?

Reply:
And I suppose we've got the same with potential for `RestServiceConfig` (apiHost / apiKey / etc)

Reply:
> Feel like there's an opportunity to create a base config that includes 'endpoint' and 'apiKey' as well. `GraphQLServiceConfig`?

I wanted to do this, but the issue is that GraphQLRequestClient's signature wants to receive endpoint and apikey separately, so it didnt seem right to define a { apiKey, endpoint } config in the client. But I'm certainly open to consider something like that. Let's save for another PR tho.